common/src/auth/ServiceConnectionString.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Text;
+
+namespace Microsoft.Azure.Devices.Common.Authorization
+{
+    /// <summary>
+    /// This object handles the connection string for the Azure IoT Services.
+    /// </summary>
+    /// <remarks>
+    /// The connection string contains a set of information that uniquely identify an IoT Service.
+    ///
+    /// A valid connection string shall be in one of the following formats:
+    /// <code>
+    /// HostName=[ServiceName];SharedAccessKeyName=[keyName];SharedAccessKey=[Key]
+    /// HostName=[ServiceName];SharedAccessKeyName=[keyName];SharedAccessSignature=[Signature]
+    /// </code>
+    ///
+    /// This object parse and store the connection string. It is responsible to provide the authorization token too.
+    /// </remarks>
+    public class ServiceConnectionString
+    {
+        protected static readonly TimeSpan DefaultTokenTimeToLive = TimeSpan.FromMinutes(5);
+
+        /// <summary>
+        /// CONSTRUCOR
+        /// </summary>
+        /// <param name="parser">the <see cref="ServiceConnectionStringParser"/> with the connection string content.</param>
+        /// <exception cref="ArgumentNullException">if the provided parser is null.</exception>
+        public ServiceConnectionString(ServiceConnectionStringParser parser)
+        {
+            if (parser == null)
+            {
+                throw new ArgumentNullException(nameof(parser));
+            }
+
+            HostName = parser.HostName;
+            SharedAccessKeyName = parser.SharedAccessKeyName;
+            SharedAccessKey = parser.SharedAccessKey;
+            SharedAccessSignature = parser.SharedAccessSignatureString;
+            ServiceName = parser.ServiceName;
+            HttpsEndpoint = new UriBuilder("https", parser.HostName).Uri;
+        }
+
+        /// <summary>
+        /// The Service Name
+        /// </summary>
+        public string ServiceName
```

Comment:
Any reason not to use the inline format?

pubic string ServiceName { get; private set; }

Reply:
No reasons, it would be good to have it same elsewhere. I know DT Service Client does not have this inline

Reply:
I'd say for all new code files, adhere to the published standard. For existing files, stick to what is internally consistent. I have a personal goal to clean up the code base for code style, so we'll get it all in check shortly.